lib/modding/CModHandler.cpp

```diff
@@ -471,47 +471,63 @@ void CModHandler::afterLoad(bool onlyEssential)
 
 }
 
-void CModHandler::trySetActiveMods(std::vector<TModID> saveActiveMods, const std::map<TModID, CModVersion> & modList)
+void CModHandler::trySetActiveMods(const std::vector<std::pair<TModID, CModInfo::VerificationInfo>> & modList)
 {
-	std::vector<TModID> newActiveMods;
-
-	ModIncompatibility::ModList missingMods;
-
+	auto hasMod = [&modList](const TModID & m) -> bool
+	{
+		for(auto & i : modList)
+			if(i.first == m)
+				return true;
+		return false;
+	};
+	
 	for(const auto & m : activeMods)
 	{
-		if (vstd::contains(saveActiveMods, m))
+		if(hasMod(m))
 			continue;
 
-		auto & modInfo = allMods.at(m);
-		if(modInfo.checkModGameplayAffecting())
-			missingMods.emplace_back(m, modInfo.version.toString());
+		if(getModInfo(m).checkModGameplayAffecting())
+			allMods[m].setEnabled(false);

```

Comment:
This thing is broken. This is how it was back in 1.2 and we had constant crashes on changing mods

VCMI Client currently can not enable or disable mod in runtime. Client does not updates own VLC when loading save and does not reconfigures filesystem to allow access to mod files

Reply:
well, it's not broken, but unused. We have same in 1.3 too, so I prefer to keep setEnabled/setDisabled methods in order to have complete logic inside this method.

Reloading VLC based on the new mod list is quite big and risky change, so I'm not going to resolve it now.